client/promotions/spotlight/index.tsx

```diff
@@ -53,24 +54,82 @@ const SpotlightPromotion: React.FC = () => {
 
 	// Find the first available promotion with a 'spotlight' variation
 	let spotlightVariation: PromotionVariation | null = null;
-	let promotionId: string | null = null;
+	let activePromotion: Promotion | null = null;
 
 	for ( const promotion of promotions ) {
 		const variation = promotion.variations.find(
 			( v ) => v.type === 'spotlight'
 		);
 		if ( variation ) {
 			spotlightVariation = variation;
-			promotionId = promotion.promo_id;
+			activePromotion = promotion;
 			break;
 		}
 	}
 
 	// No spotlight promotion available
-	if ( ! spotlightVariation || ! promotionId ) {
+	if ( ! spotlightVariation || ! activePromotion ) {
 		return null;
 	}
 
+	// Extract values after null check for TypeScript
+	const promotionId = activePromotion.promo_id;
+	const paymentMethod = activePromotion.payment_method;
+	const variationId = spotlightVariation.id;
+	const ctaLabel = spotlightVariation.cta_label;
+	const ctaUrl = spotlightVariation.cta_url;
+
+	/**
+	 * Get common event properties for tracking.
+	 */
+	const getEventProperties = () => ( {
+		promotion_id: promotionId,
+		payment_method: paymentMethod,
+		variation_id: variationId,
+		display_context: 'spotlight',
+		page: window.location.href,
```

Comment:
This is not very helpful since it is the full URL. We need to better determine on which screen we are. At the very least we should only include the path part (like the [wcadmin_page_view](https://mc.a8c.com/tracks/events/event.php?eventname=wcadmin_page_view) event is doing)